### ISSUE TITLE
🐛 noUselessElse has a unhelpful/unclear description

### ISSUE THREAD

**Issue by net-tech** [biomejs/biome#920]
### Environment information

```block
CLI:
  Version:                      1.4.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.4.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

Currently, the description for the `noUselessElse` rule reads 

> Disallow else block when the if block breaks early.
If an if block breaks early using a breaking statement (return, break, continue, or throw), then the else block becomes useless. Its contents can be placed outside of the block.

which is not very clear in explaining why the `else` block becomes useless. This could result in people who cannot interpret the reasoning of why the else block becomes useless to disable the rule.

### Expected result

A better alternative could be something like this

> If an if block breaks early using a breaking statement (return, break, continue, or throw), the else block becomes unnecessary. This is because the contents of the else block will never be executed in conjunction with the if block, as the breaking statement ensures the control flow exits the if block immediately. Therefore, the else block is redundant, and its contents can be more effectively placed outside of the block, enhancing clarity and efficiency in the code structure.

Let me know if I should modify anything or make a PR for it.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by nstepien** [comment]
>Therefore, the else block is redundant, and its contents can be more effectively placed outside of the block, enhancing clarity and efficiency in the code structure.

```js
if (a) {
  return ...;
} else if (b) {
  return ...;
}
```

Here the `else` is redundant, but the condition is not, so the "block" should remain:

```js
if (a) {
  return ...;
}

if (b) {
  return ...;
}
```

I think rather than talking about "blocks", the description should talk about the `else` clause itself.
